App.js

```diff
@@ -40,19 +44,43 @@ class App extends Component{
     this.fetchApiData();
   }
 
+  sortDataById(){
+    const newData = this.state.data.sort((a,b) => a.id - b.id);
+    this.setState({
+      data: newData,
+    });
+  }
+  
+  sortDataByAuthor(){
+    const newData = this.state.data.sort((a,b) => {
+      if(a.author > b.author) return 1;
+      else if(a.author < b.author) return -1;
+      else return 0;
+    });
+    this.setState({
+      data: newData,
+    });
+  }
+
   render(){
     const body = this.state.isLoading
     ? <View style={styles.loadingIndicator}>
         <ActivityIndicator size='large' color='#058cd9' show={this.state.isLoading}/>
       </View>
     : <FlatList
         data = {this.state.data}
+        extraData = {this.state}
```

Comment:
Why u use this property ? list depend only on `state.data` so for sure you don't need to do that